### PR TITLE
Move examples to their own subdirectories

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Move examples to their own subdirectories.

Ref: #427

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
Ref: #427

**Review by KFilipek — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -2,73 +2,14 @@
...
-    set_tests_properties(${EXAMPLE_NAME} PROPERTIES LABELS "example")
-
-    if(WINDOWS)
-        # append PATH to DLLs
-        set_property(TEST ${EXAMPLE_NAME} PROPERTY ENVIRONMENT_MODIFICATION
-                                                   "${DLL_PATH_LIST}")
-    endif()
-endif()
+# basic example
+add_subdirectory(basic)
```
[the remark this answers is not in the document]
Comment (lplewa) [reply examples/CMakeLists.txt:6]: Well this is "ref" the issue, not "fixes it"

[the remark this answers is not in the document]
Comment (ldorau) [reply examples/CMakeLists.txt]: @lukaszstolarczuk Reverted for now - it will be fixed in a separate PR.

[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply examples/CMakeLists.txt]: thx!

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**